psl/psl-core/src/configuration/generator.rs

```diff
@@ -1,15 +1,41 @@
 use crate::{configuration::StringFromEnvVar, PreviewFeature};
 use enumflags2::BitFlags;
+use parser_database::ast::Expression;
 use serde::{ser::SerializeSeq, Serialize, Serializer};
 use std::collections::HashMap;
 
+#[derive(Debug, Serialize, Clone)]
+#[serde(untagged)]
+pub enum GeneratorConfigValue {
+    String(String),
+    Array(Vec<GeneratorConfigValue>),
+}
+
+impl From<String> for GeneratorConfigValue {
+    fn from(value: String) -> Self {
+        Self::String(value)
+    }
+}
+
+impl From<&Expression> for GeneratorConfigValue {
+    fn from(expr: &Expression) -> Self {
+        match expr {
+            Expression::NumericValue(val, _) => val.clone().into(),
+            Expression::StringValue(val, _) => val.clone().into(),
+            Expression::ConstantValue(val, _) => val.clone().into(),
+            Expression::Function(_, _, _) => "(function)".to_owned().into(),
+            Expression::Array(elements, _) => Self::Array(elements.iter().map(From::from).collect()),
```

Comment:
So this means we would supported nested arrays. We should have a quick test for it.